### PR TITLE
Final steps towards rel.

### PR DESCRIPTION
**Pull request by JulianFeinauer**
Removed "build" module and moved this to parent pom.
Renamed parent pom without incubating.
Added incubating artefact names in assembly plugin.
Added SHA512 for all artefacts.

Big props to @chrisdutz for his support!

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -584,5 +583,78 @@
...
+                                <goals>
+                                    <goal>single</goal>
+                                </goals>
+                                <!-- heads up: combine.self in the following is highlighted
+                                    as an error in Eclipse's xml editor view.
+                                    Just ignore that.
+                                    See  https://issues.apache.org/jira/browse/MNG-5454  sigh.
+                                 -->
+                                <configuration combine.self="append">
+                                    <finalName>apache-iotdb-${project.version}-incubating</finalName>
```
Comment (jixuan1989) [inline pom.xml]: Oh, now I get what Justin means... just add "incubating" here is fine..

Comment (JulianFeinauer) [reply pom.xml]: yeah.. therefore I changed the name of the module back to "iotdb-parent" which is more standard : )

**Comment by jixuan1989** [comment]
Hi, I tried this branch but after running `mvn package`, I do not get the .zip file...

**Comment by JulianFeinauer** [comment]
@jixuan1989 you have to use `mvn package -Papache-release` to enable these steps.

**Pull request merged**